v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,466 @@
+
+        if (colValue instanceof String) {
+            try {
+                date = new SimpleDateFormat(formatter).parse((String) colValue);
+            } catch (ParseException e) {
+                throw new IllegalArgumentException("Invalid timestamp format for column " + colName, e);
+            }
+        } else if (colValue instanceof java.util.Date) {
+            date = (java.util.Date) colValue;
+        } else if (colValue instanceof Long) {
+            date = new Date((Long) colValue);
+        } else {
+            throw new IllegalArgumentException("Unsupported type for column " + colName);
+        }
+
+        return date;
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link String} object containing String as value represented in cassandra type.
+     */
+    public static String handleCassandraTextType(String colName, JSONObject valuesJson) {
+        return valuesJson.optString(colName, null); // Get the value or null if the key is not found or the value is null
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link UUID} object containing UUID as value represented in cassandra type.
+     */
+    public static UUID handleCassandraUuidType(String colName, JSONObject valuesJson) {
+        String uuidString = valuesJson.optString(colName, null); // Get the value or null if the key is not found or the value is null
+
+        if (uuidString == null) {
+            return null;
+        }
+
+        return UUID.fromString(uuidString);
+
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Long} object containing Long as value represented in cassandra type.
+     */
+    public static Long handleCassandraBigintType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigInteger(colName).longValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Integer} object containing Integer as value represented in cassandra type.
+     */
+    public static Integer handleCassandraIntType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigInteger(colName).intValue();
```

Comment:
In case we are using Cassandra driver to write, atleast on the read path, I have experienced it doing strong type checks for `tinyInt` and `MediumInt` with java `Byte` and `Short`.
On Spanner side all of the smaller integer version will map to Integer alone.
It might make sense to add separate handling for such types, and in case of possibility of truncation, log error/ Sever DLQ entries.

Reply:
hi @VardhanThigle We are converting the value coming from Spanner to Integer and while assigning we are converting to Byte and Short under Typehandler mapping getColumnValueByType
<img width="655" alt="Screenshot 2024-12-10 at 11 47 25 AM" src="https://github.com/user-attachments/assets/6c125d93-563f-4a34-b201-c6d7b87b49f6">

Reply:
Thanks. Is this snippet from a future PR?
Would it make sense to put all the conversions at one place?
As in what if we try to convert it to `short` or `byte` in this class itself?
That way all the conversions needed to convert from Spanner to Cassandra compatible types will be in a single place (to the extent possible).

Reply:
@VardhanThigle  we are doing in the same class  which will be provide more visibility in further PR

Reply:
@shreyakhajanchi We have added the rest datatypes as well in the same PR